pkg/gitnessclient/gitness_client.go

```diff
@@ -0,0 +1,416 @@
+	if err != nil {
+		return nil, fmt.Errorf("failed to perform request: %w", err)
+	}
+	defer resp.Body.Close()
+
+	if resp.StatusCode != http.StatusOK {
+		return nil, fmt.Errorf("unexpected status code: %d", resp.StatusCode)
+	}
+
+	body, err := io.ReadAll(resp.Body)
+	if err != nil {
+		return nil, fmt.Errorf("failed to read response body: %w", err)
+	}
+	var apiUser apiUserResponse
+	if err := json.Unmarshal(body, &apiUser); err != nil {
+		return nil, fmt.Errorf("failed to unmarshal response body: %w", err)
+	}
+
+	return &apiUser, nil
+}
+
+func (g *GitnessClient) GetRepositories(namespace string) ([]ApiRepository, error) {
+	space := ""
+	if namespace == personalNamespaceId {
+		user, err := g.GetUser()
+		if err != nil {
+			return nil, err
+		}
+		space = user.UID
+	} else {
+		space = namespace
+	}
+
+	reposURL, err := g.BaseURL.Parse(fmt.Sprintf("/api/v1/spaces/%s/+/repos", space))
+	if err != nil {
+		return nil, fmt.Errorf("failed to parse url : %w", err)
+	}
+
+	req, err := http.NewRequest("GET", reposURL.String(), nil)
+	if err != nil {
+		return nil, err
+	}
+
+	req.Header.Set("accept", "application/json")
+	req.Header.Set("Authorization", "Bearer "+g.Token)
+
+	client := &http.Client{}
+	resp, err := client.Do(req)
+	if err != nil {
+		return nil, err
+	}
+	defer resp.Body.Close()
+
+	if resp.StatusCode != http.StatusOK {
+		return nil, fmt.Errorf("failed to fetch repositories, status code: %d", resp.StatusCode)
+	}
+
+	body, err := io.ReadAll(resp.Body)
+	if err != nil {
+		return nil, err
+	}
+
+	var apiRepos []ApiRepository
+	if err := json.Unmarshal(body, &apiRepos); err != nil {
+		return nil, err
+	}
+
+	return apiRepos, nil
+}
+
+func (g *GitnessClient) GetRepoBranches(repositoryId string, namespaceId string) ([]*apiRepoBranch, error) {
+	branchesURL, err := g.BaseURL.Parse(fmt.Sprintf("/api/v1/repos/%s/branches", url.PathEscape(namespaceId+"/"+repositoryId)))
+	if err != nil {
+		return nil, fmt.Errorf("failed to parse url : %w", err)
+	}
+
+	req, err := http.NewRequestWithContext(context.Background(), "GET", branchesURL.String(), nil)
+	if err != nil {
+		return nil, fmt.Errorf("failed to create request: %w", err)
+	}
+
+	req.Header.Set("accept", "application/json")
+	req.Header.Set("Authorization", "Bearer "+g.Token)
+
+	client := &http.Client{}
+	resp, err := client.Do(req)
+	if err != nil {
+		return nil, fmt.Errorf("failed to perform request: %w", err)
+	}
+	defer resp.Body.Close()
+
+	if resp.StatusCode != http.StatusOK {
+		return nil, fmt.Errorf("unexpected status code: %d", resp.StatusCode)
+	}
+
+	body, err := io.ReadAll(resp.Body)
+	if err != nil {
+		return nil, fmt.Errorf("failed to read response body: %w", err)
+	}
+
+	var branches []*apiRepoBranch
+	if err := json.Unmarshal(body, &branches); err != nil {
+		return nil, fmt.Errorf("failed to unmarshal response body: %w", err)
+	}
+
+	return branches, nil
+}
+
+func (g *GitnessClient) GetRepoPRs(repositoryId string, namespaceId string) ([]*apiPR, error) {
```

Comment:
![Screenshot 2024-06-19 at 09 48 31](https://github.com/daytonaio/daytona/assets/26512078/81a7f11b-91c1-43fe-afd2-4666d8287619)

Not sure because of what exactly does this issue occur but leaving it here. The screenshot in question was taken after creating a workspace from a PR with `dtn create` (using the TUI).

Everything seems to work great when I create with `dtn create PR_URL (e.g. http://localhost:3000/test/test/pulls/1`) so I assume it has something to do with this PR list.